dipy/reconst/csdeconv.py

```diff
@@ -495,8 +495,12 @@ def odf_deconv(odf_sh, R, B_reg, lambda_=1., tau=0.1, r2_term=False):
         return np.zeros_like(odf_sh), 0
 
     # Generate initial fODF estimate, which is the ODF truncated at SH order 4
-    fodf_sh = np.linalg.lstsq(R, odf_sh)[0]
-    fodf_sh[15:] = 0
+    try:
+        fodf_sh = np.linalg.lstsq(R, odf_sh)[0]
```

Comment:
If we re-factor this function to use cholesky decomposition or psudo-inverse we should be able to avoid this error because the decomposition/inverse is data-independent. cholesky decomposition might also improve performance based on what we're seeing in the CSD code.